test/tests.ts

```diff
@@ -2167,86 +2229,90 @@ describe("passport-saml /", function () {
-        var samlConfig = {
+        const samlConfig = {
           entryPoint: "",
           issuer: "acme_tools_com",
           callbackUrl: "https://relyingparty/adfs/postResponse",
           privateKey: fs.readFileSync(__dirname + "/static/acme_tools_com.key", "utf-8"),
           authnContext:
             "http://schemas.microsoft.com/ws/2008/06/identity/authenticationmethod/password",
-          signatureAlgorithm: "sha256",
+          signatureAlgorithm: "sha256" as const,
           additionalParams: {
             customQueryStringParam: "CustomQueryStringParamValue",
           },
         };
-        var samlObj = new SAML(samlConfig);
+        const samlObj = new SAML(samlConfig);
         samlObj.generateUniqueID = function () {
           return "12345678901234567890";
         };
 
-        var request =
+        const request =
           '<?xml version=\\"1.0\\"?><samlp:AuthnRequest xmlns:samlp=\\"urn:oasis:names:tc:SAML:2.0:protocol\\" ID=\\"_ea40a8ab177df048d645\\" Version=\\"2.0\\" IssueInstant=\\"2017-08-22T19:30:01.363Z\\" ProtocolBinding=\\"urn:oasis:names$tc:SAML:2.0:bindings:HTTP-POST\\" AssertionConsumerServiceURL=\\"https://example.com/login/callback\\" Destination=\\"https://www.example.com\\"><saml:Issuer xmlns:saml=\\"urn:oasis:names:tc:SAML:2.0:assertion\\">onelogin_saml</saml:Issuer><s$mlp:NameIDPolicy xmlns:samlp=\\"urn:oasis:names:tc:SAML:2.0:protocol\\" Format=\\"urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress\\" AllowCreate=\\"true\\"/><samlp:RequestedAuthnContext xmlns:samlp=\\"urn:oasis:names:tc:SAML:2.0:protoc$l\\" Comparison=\\"exact\\"><saml:AuthnContextClassRef xmlns:saml=\\"urn:oasis:names:tc:SAML:2.0:assertion\\">urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport</saml:AuthnContextClassRef></samlp:RequestedAuthnContext></samlp$AuthnRequest>';
         samlObj.requestToUrl(request, null, "authorize", {}, function (err) {
           try {
             should.exist(err);
-            err.message.should.eql('"entryPoint" config parameter is required for signed messages');
+            err!.message!.should.eql(
+              '"entryPoint" config parameter is required for signed messages'
+            );
             done();
           } catch (err2) {
             done(err2);
           }
         });
       });
       it("acme_tools request signed with sha1", function (done) {
-        var samlConfig = {
+        const samlConfig = {
           entryPoint: "https://adfs.acme_tools.com/adfs/ls/",
           issuer: "acme_tools_com",
           callbackUrl: "https://relyingparty/adfs/postResponse",
           privateCert: fs.readFileSync(__dirname + "/static/acme_tools_com.key", "utf-8"),
           authnContext:
             "http://schemas.microsoft.com/ws/2008/06/identity/authenticationmethod/password",
           identifierFormat: null,
-          signatureAlgorithm: "sha1",
+          signatureAlgorithm: "sha1" as const,
```

Comment:
You shouldn't need `as const` here.

Reply:
we need it, because if we don't add it here, it sets type as string and string is too broad for type 'sha1' | 'sha256‘ | ... used in the module

Reply:
I understand what you are saying now... in that case, you can change it from `as const` to `as SignatureAlgorithm`. In fact, I've just pushed that change up.